src/main/java/com/empowerops/linqalike/Linq.java

```diff
@@ -636,16 +662,23 @@ TElement withMax(Queryable<TElement> sourceElements,
         return ImmediateInspections.indexOf(sourceElements, elementToFind, equalityComparer);
     }
 
+
     public static <TElement> int lastIndexOf(Iterable<TElement> sourceElements,
                                              TElement elementToFind,
                                              EqualityComparer<? super TElement> equalityComparer) {
         return ImmediateInspections.indexOf(reversed(sourceElements), elementToFind, equalityComparer);
     }
 
-    public static <TElement, TCompared> int indexOf(Iterable<? extends TElement> sourceElements,
-                                                    TElement elementToFind,
-                                                    Func1<? super TElement, TCompared> comparableSelector) {
-        return ImmediateInspections.indexOf(sourceElements, elementToFind, performEqualsUsing(memoizedSelector(comparableSelector)));
+    @SuppressWarnings("unchecked") //cast of a dynamically assured type
+    public static <TLeft, TSubclassLeft extends TLeft, TRight>
+    BiQueryable<TSubclassLeft, TRight> ofLeftType(BiQueryable<TLeft, TRight> tuples, Class<TSubclassLeft> desiredLeftClass) {
+        return (BiQueryable) where(tuples, tuple -> desiredLeftClass.isInstance(tuple.left));
+    }
+
+    @SuppressWarnings("unchecked") //cast of a dynamically assured type
+    public static <TLeft, TSubclassRight extends TRight, TRight>
+    BiQueryable<TLeft, TSubclassRight> ofRightType(BiQueryable<TLeft, TRight> tuples, Class<TSubclassRight> desiredRightClass) {
+        return (BiQueryable) where(tuples, tuple -> desiredRightClass.isInstance(tuple.right));
```

Comment:
am I doing this kind of interception and conversion anywhere else? I think I was just too lazy to write a static method